rx_signal_processing/decimate.hpp

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2017 SuperDARN Canada
+
+See LICENSE for details
+  \file decimate.hpp
+*/
+
+#ifndef DECIMATE_H
+#define DECIMATE_H
+
+#include "dsp.hpp"
+
+enum class DecimationType {lowpass, bandpass};
+
+void bandpass_decimate1024_wrapper(cuComplex* original_samples,
+  cuComplex* decimated_samples,
+  cuComplex* filter_taps, uint32_t dm_rate,
+  uint32_t samples_per_antenna, uint32_t num_taps_per_filter, uint32_t num_freqs,
+  uint32_t num_antennas, cudaStream_t stream);
+
+void bandpass_decimate2048_wrapper(cuComplex* original_samples,
+  cuComplex* decimated_samples,
+  cuComplex* filter_taps, uint32_t dm_rate,
+  uint32_t samples_per_antenna, uint32_t num_taps_per_filter, uint32_t num_freqs,
+  uint32_t num_antennas, cudaStream_t stream);
+
+void lowpass_decimate1024_wrapper(cuComplex* original_samples,
+  cuComplex* decimated_samples,
+  cuComplex* filter_taps, uint32_t dm_rate,
+  uint32_t samples_per_antenna, uint32_t num_taps_per_filter, uint32_t num_freqs,
+  uint32_t num_antennas, cudaStream_t stream);
+
+void lowpass_decimate2048_wrapper(cuComplex* original_samples,
+  cuComplex* decimated_samples,
+  cuComplex* filter_taps, uint32_t dm_rate,
+  uint32_t samples_per_antenna, uint32_t num_taps_per_filter, uint32_t num_freqs,
+  uint32_t num_antennas, cudaStream_t stream);
+/**
+ * @brief      Selects which decimate kernel to run.
+ *
+ * @param[in]  original_samples     A pointer to original input samples from
+ *                                  each antenna to decimate.
+ * @param[in]  decimated_samples    A pointer to a buffer to place output
+ *                                  samples for each frequency after decimation.
+ * @param[in]  filter_taps          A pointer to one or more filters needed for
```

Comment:
could you explain that in the comment that lowpass is a single filter, bandpass can be multiple depending on number of frequencies.